arch/risc-v/src/common/riscv_addrenv.c

```diff
@@ -187,6 +193,38 @@ static int create_spgtables(group_addrenv_t *addrenv)
   return i;
 }
 
+/****************************************************************************
+ * Name: copy_kernel_mappings
+ *
+ * Description:
+ *   Copy kernel mappings to address environment. Expects that the user page
+ *   table does not contain any mappings yet (as they will be wiped).
+ *
+ * Input Parameters:
+ *   addrenv - Describes the address environment. The page tables must exist
+ *      at this point.
+ *
+ * Returned value:
+ *   OK on success, ERROR on failure
+ *
+ ****************************************************************************/
+
+static int copy_kernel_mappings(group_addrenv_t *addrenv)
+{
+  uintptr_t user_mappings = addrenv->spgtables[0];
+
+  /* Copy the L1 references */
+
+  if (user_mappings == 0)
+    {
+      return ERROR;
+    }
+
+  memcpy((void *)user_mappings, (void *)g_kernel_mappings, ENTRIES_PER_PGT);
```

Comment:
should we copy byte size instead entries(ENTRIES_PER_PGT)?

Reply:
Of course, the entire table needs to be copied.